### PR TITLE
Merge block-stm to develop

### PR DESCRIPTION
**Pull request by cffls**
# Description

Enables parallel execution of EVM.

# Changes

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)
- [ ] Changes only for a subset of nodes

# Breaking changes

Please complete this section if any breaking changes have been made, otherwise delete it

# Nodes audience

In case this PR includes changes that must be applied only to a subset of nodes, please specify how you handled it (e.g. by adding a flag with a default value...)

# Checklist

- [ ] I have added at least 2 reviewer or the whole pos-v1 team
- [ ] I have added sufficient documentation in code
- [ ] I will be resolving comments - if any - by pushing each fix in a separate commit and linking the commit hash in the comment reply

# Cross repository changes

- [ ] This PR requires changes to heimdall
    - In case link the PR here:
- [ ] This PR requires changes to matic-cli
    - In case link the PR here:

## Testing

- [ ] I have added unit tests
- [ ] I have added tests to CI
- [ ] I have tested this code manually on local environment
- [ ] I have tested this code manually on remote devnet using express-cli
- [ ] I have tested this code manually on mumbai
- [ ] I have created new e2e tests into express-cli

### Manual tests

Please complete this section with the steps you performed if you ran manual tests for this functionality, otherwise delete it

# Additional comments

Please post additional comments in this section if you have them, otherwise delete it

### CODE REVIEW

File: cmd/utils/flags.go [hunk 1]
```diff
@@ -2084,6 +2084,7 @@ func MakeChain(ctx *cli.Context, stack *node.Node) (chain *core.BlockChain, chai
 
 	// TODO(rjl493456442) disable snapshot generation/wiping if the chain is read only.
 	// Disable transaction indexing/unindexing by default.
+	// PSP - Check for config.ParallelEVM.Enable here?
```
Comment (marcello33) [inline cmd/utils/flags.go:2087]: Are those reminders for next iterations?

**Review by marcello33 — APPROVED** [review]

File: cmd/utils/flags.go (hunk 1 above)
Comment (cffls) [reply cmd/utils/flags.go]: Nice catch! The todo is not required anymore. Will remove this comment.

Comment (pratikspatil024) [reply cmd/utils/flags.go]: [Removed](https://github.com/maticnetwork/bor/pull/840/commits/adce0e823527da26822a4076e1a977c6bc6e6d00).

**Review by pratikspatil024 — APPROVED** [review]

File: docs/config.md
[the remark this answers is not in the document]
Comment (pratikspatil024) [reply docs/config.md:1]: [Removed](https://github.com/maticnetwork/bor/pull/840/commits/0a7594b17ac872328806afb9c9e7181e3b5e5a34).

**Pull request merged**